tutorials/asr/cloud/aws/SageMaker_ASR_Fine-tuning-CommonVoice.ipynb

```diff
@@ -0,0 +1,666 @@
+   "source": [
+    "config_path = str(config_dir / \"config.yaml\")\n",
+    "wget.download(\"https://raw.githubusercontent.com/andrusenkoau/NeMo/esperanto_example/docs/source/asr/examples/esperanto_asr/scripts/get_commonvoice_data_v2.py\", str(code_dir))\n",
+    "wget.download(\"https://raw.githubusercontent.com/NVIDIA/NeMo/v1.13.0/examples/asr/conf/conformer/conformer_ctc_bpe.yaml\", config_path)\n",
+    "wget.download(\"https://raw.githubusercontent.com/NVIDIA/NeMo/v1.13.0/examples//asr/asr_ctc/speech_to_text_ctc_bpe.py\",\n",
+    "              str(code_dir))\n",
+    "wget.download(\"https://raw.githubusercontent.com/NVIDIA/NeMo/v1.13.0/scripts/tokenizers/process_asr_text_tokenizer.py\",\n",
+    "              str(code_dir))\n",
+    "wget.download(\"https://raw.githubusercontent.com/NVIDIA/NeMo/v1.13.0/scripts/speech_recognition/convert_to_tarred_audio_dataset.py\",\n",
+    "              str(code_dir))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7934baab",
+   "metadata": {},
+   "source": [
+    "We also create a `requirements.txt` file within our source code to install NeMo."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "397d8eef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open(code_dir / 'requirements.txt', 'w') as f:\n",
+    "    f.write(\"nemo_toolkit[all]\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a7bc7f51",
+   "metadata": {},
+   "source": [
+    "### 2.1 Install Binaries within Training Script\n",
+    "\n",
+    "We install `sndfile` and `ffmpeg` within our python training script. These are required system libraries that do not come with the pre-built SageMaker container.\n",
+    "\n",
+    "By installing it on the fly we save having to import a custom container, and can leverage all of the SageMaker containers. Rather than running this cell, you could also manually do this in your script."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e17535a7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# prepend system call to training script to install libraries\n",
+    "line = \"import os; os.system('chmod 777 /tmp && apt-get update && apt-get install -y libsndfile1 ffmpeg');\"\n",
+    "with open(code_dir / \"speech_to_text_ctc.py\", 'r+') as f:\n",
+    "    content = f.read()\n",
+    "    f.seek(0, 0)\n",
+    "    f.write(line.rstrip('\\r\\n') + '\\n' + content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ce3e5ae8",
+   "metadata": {},
+   "source": [
+    "### 3. Setup the CommonVoice Esperanto Dataset\n",
+    "\n",
+    "We run the download script, which downloads the subset of the dataset we'll be training/evaluating on. Note this takes a few hours to download."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cadd2382",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python ./code/get_commonvoice_data_v2.py \\\n",
```

Comment:
Probably better to go with the MCV + HF sanctioned route here than the internal script that directly access the data. MCV does not like that. It will require the user providing their HF credentials

Reply:
We have a script for downloading data from HF and it creates a temp manifest which we then update to have NeMo required keys (audio_filepath, text, duration)

Reply:
@vsl9 please advice

Reply:
Spoke to @vsl9 offline, we should not be using the internal script, will move towards the HF approach!